python/ray/tune/ray_trial_executor.py

```diff
@@ -157,6 +157,14 @@ def pause_trial(self, trial):
             self._paused[trial_future[0]] = trial
         super(RayTrialExecutor, self).pause_trial(trial)
 
+    def reset_trial(self, trial, new_config):
+        """If trainable (trial) supports resetting config on the fly,
+        reset config and return True, otherwise return False."""
+        if hasattr(trial, 'reset_config') and callable(trial.reset_config):
```

Comment:
Hmm I'm not sure this will detect if its overriden properly. How about calling reset_config(), then catching NotImplementedError and falling back to stop/start?

Reply:
+1 on catching the NotImplemented